### DISCUSSION TITLE
Is there a good reason why Author ships fine you for illegal outfits?

### DISCUSSION THREAD

**Discussion by brendanjones** [endless-sky/endless-sky#9974] [Balance, Mechanics, and Interactions]
It's just plain annoying. It reminds me of Capt'n Hector in original EV attacking you for not paying for the game, except there's no reason for it here. I would gladly see this mechanic removed.

It's especially annoying given some of the Author ships are super fast and will come and check you out no matter where you are in the system, meaning you can't avoid them scanning you, unlike other NPC ships.

**Comment by warp-core** [comment]
Cap'n Pester is a direect reference to Cap'n Hector.
Of the others, I think it's Local God, Zitchas, MasterOfGrey, and Patrol Team that can scan your outfits or cargo.
For Zitchas and MasterOfGrey, the normal ships the person ships are based on have the scanning capabilities.
I can't say why the others have it.

**Reply by brendanjones** [reply]
Aye, but considering author ships can be custom there’s nothing stopping us taking out the scanners. So is the reason then “it’s unintentional, because we didn’t take the scanners out”?

If there’s no actual reason for it then I’d argue to remove them.

**Reply by MasterOfGrey** [reply]
In an ideal world mine would just send you a message to the effect of, "lol, don't let the law catch you with those."

But if the author government is always going to fine you because of the Cap'n Pester behaviour then mine should have the outfit scanning abilities removed.
(any other scanners should stay)

**Reply by mOctave** [reply]
https://github.com/endless-sky/endless-sky/pull/9983

**Reply by brendanjones** [reply]
Ah, https://github.com/endless-sky/endless-sky/issues/6747 wasn’t accepted. I really ought to search closed PRs…

I assume tehhowch is a core dev, given his comment and then adding the wontfix. https://github.com/endless-sky/endless-sky/issues/6747#issuecomment-1103889300

**Reply by Saugia** [reply]
Yes, he is. The idea is that Author ships should be able to do anything others can. So I think it's fine we don't restrict this, since the ships were built with these intentions at the time, and it would feel odd to suddenly change them.

As a side note, yes, Capn' Pester is intentionally designed to be like Hecter from EV, but I think there's a huge difference between attacking the player and scanning them.

What I will say is that person ship spawn rates are a lot higher than we generally would like them at the moment. I plan on going deeper into that and pulling them back a notable amount eventually.

**Reply by brendanjones** [reply]
> since the ships were built with these intentions at the time, and it would feel odd to suddenly change them.

Were they and would it? Master of Grey's comments above and the resulting PR would indicate not. In any case the author ships are not part of the 'proper' game experience, they already have ships/tech and behave in ways that don't otherwise exist in game, so I don't think changing their scanning behaviour is much of a problem.

> What I will say is that person ship spawn rates are a lot higher than we generally would like them at the moment. I plan on going deeper into that and pulling them back a notable amount eventually.

Definitely agree with this, considering some of them are a sort of spoiler (e.g. Zitchas' Heron), and sort of easter eggs. They appear quite commonly! They're also annoying when you're trying to disable ships and they come in and destroy everything. Especially early game when you've spent a bunch of time disabling a ship and then an overpowered author ship comes in a wrecks your work (bloody Tehhowch and Zitchas lol).

So, if their spawn rate is reduced then I would be less bothered by the scanning remaining on the author ships that have it, and I'll leave my complaint as 'noted' and hope for the spawn rate to get reduced at some point :)

**Reply by Saugia** [reply]
Grey's person ship was made a lot more recently than the others. The original pers ship group came out with the game's release, since then, a few have been added, but there is a distinction between the persons of original developers of the game's creation and contributors since then (for example, we avoid any new persons being evenly compared with them or something to that effect).

The ladybug itself comes with scanning, and Grey not wanting for it to interfere with this is understandable and an aside. However the original pers ships are a bit of a different story and came with the game as they were from the original crew, so it's safe to say that they were intended and knowingly made this way.

Yes, I feel spawn rates will fix this and some other tangential issues (per ships appearing often in distant uninhabited systems, etc.).

**Reply by ziproot** [reply]
If the issue is Cap‘n Pester, she already has her own government that is separate from the other person ships, so the Author government can still be given a fine 0 without it applying to Cap’n Pester. I also don’t really get tehhowch’s argument applying to fines, as there are already ships in the game that scan but don’t fine you. They can fine you, they just choose not to.

EDIT: I personally see the Author ships as other players, who therefore could scan but not fine in the same way that you can’t fine other people.

**Reply by TheMarksman-ES** [reply]
Patrol Team's scanning was added because of the stuff that we do here (which is scanning through a PR for errors and mistakes and what not.) It's also likely the most annoying Person ship fleet, given that there's a good chance that'd you'd be fined five times more than any other author ship.
